app/scripts/metamask-controller.js

```diff
@@ -1475,15 +1476,24 @@ export default class MetamaskController extends EventEmitter {
         },
         messenger: this.controllerMessenger.getRestricted({
           name: 'TransactionController',
-          allowedActions: [`${this.approvalController.name}:addRequest`],
+          allowedActions: [
+            `${this.approvalController.name}:addRequest`,
+            'NetworkController:getNetworkClientById',
+            'NetworkController:findNetworkClientIdByChainId',
+          ],
+          allowedEvents: [`NetworkController:stateChange`],
         }),
         onNetworkStateChange: (listener) => {
           networkControllerMessenger.subscribe(
             'NetworkController:networkDidChange',
             () => listener(),
           );
         },
-        provider: this.provider,
+        isMultichainEnabled: process.env.TRANSACTION_MULTICHAIN,
```

Comment:
Minor, but we're trying to keep these in alphabetical order to help with maintainability, expect for `hooks` at the bottom.

Reply:
fixed here https://github.com/MetaMask/metamask-extension/pull/22979/commits/7ea2fee6cddda294c73a6e79119f9818248badd3